src/psyclone/transformations.py

```diff
@@ -412,89 +410,79 @@ def validate_it_can_run_on_gpu(self, node, options):
             # or that the frontend failed to convert it into PSyIR) reraise it
             # as a TransformationError
             try:
-                kernel_schedule = node.get_kernel_schedule()
+                kernel_schedules = node.get_kernel_schedule()
             except Exception as error:
                 raise TransformationError(
                     f"Failed to create PSyIR for kernel '{node.name}'. "
                     f"Cannot transform such a kernel.") from error
 
-            if not node.ancestor(Container, shared_with=kernel_schedule):
-                # The KernelSchedule to be transformed has not been inlined
-                # into the Container of the call-site. Therefore we can
-                # Check that it's not a mixed-precision kernel (which will have
-                # more than one Routine implementing it). We can't transform
-                # these at the moment because we can't correctly manipulate
-                # their metadata - TODO #1946.
-                if len(kernel_schedule.root.walk(Routine)) > 1:
-                    raise TransformationError(
-                        f"Cannot apply {self.name} to kernel '{node.name}' as "
-                        f"it has multiple implementations - TODO #1946")
-
             k_or_r = "Kernel"
         else:
             # Supplied node is a PSyIR Routine which *is* a Schedule.
-            kernel_schedule = node
+            kernel_schedules = [node]
             k_or_r = "routine"
```

Comment:
This also seems to take a kern (call) or a routine. Nemo needs routines as we search them by name since they may not be in the same file than the caller. Would it be possible to bring the `kernel_schedules = node.get_kernel_schedule()` outside in LFRic where this is called to be consistent? We can still do:
```
if isinstance(node, KernScedule):
```
if it helps to give different error messages.

(Can be a separate PR is this is getting to big)

Reply:
I see below that this is also the case in OMPDeclareTargetTrans, so maybe a separate PR

Reply:
Yes, I started doing this and it got out of hand. As things stand, it would become the responsibility of the person writing the transformation script to do `node.get_callees()`, allow for cases where the PSyIR can't be obtained, and then call the transformation on each result. That doesn't feel very nice.